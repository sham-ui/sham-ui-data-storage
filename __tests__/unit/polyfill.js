const { setImmediate } = window;
const updating = () => new Promise( resolve => setTimeout( resolve ) );

beforeAll( () => {
    delete window.setImmediate;
} );

afterAll( () => {
    window.setImmediate = setImmediate;
} );

import renderer, { compileAsSFC } from 'sham-ui-test-helpers';
import createStorage from '../../src/index';


it( 'with setImmediate polyfill', async() => {
    const { storage, useStorage }  = createStorage( {
        firstName: 'John',
        lastName: 'Smith'
    } );

    const meta = renderer(
        compileAsSFC( {
            useStorage
        } )`
            <template>
                <dl>
                    <dt>First name</dt>
                    <dd>{{user.firstName}}</dd>
                    <dt>Last name</dt>
                    <dd>{{user.lastName}}</dd>
                </dl>
            </template>
            <script>
                export default Component( Template, useStorage( $.user ) );
            </script>
        `
    );
    expect( meta.toJSON() ).toMatchSnapshot( 'initial' );
    const user = storage( meta.DI );
    user.firstName = 'Jordan';
    expect( meta.toJSON() ).toMatchSnapshot( 'after update firstName' );
    await updating();
    expect( meta.toJSON() ).toMatchSnapshot( 'after update firstName & wait updating' );
    user.lastName = 'Shah';
    expect( meta.toJSON() ).toMatchSnapshot( 'after update lastName' );
    expect( meta.toJSON() ).toMatchSnapshot( 'after update lastName & wait updating' );
    user.lastName = 'Shah';
    await updating();
    expect( meta.toJSON() ).toMatchSnapshot( 'lastName don\'t changed' );
} );
