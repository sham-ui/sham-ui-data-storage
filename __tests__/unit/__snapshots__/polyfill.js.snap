// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with setImmediate polyfill: after update firstName & wait updating 1`] = `
Object {
  "Options": Object {
    "ID": "component",
    "user": Object {
      "firstName": "Jordan",
      "lastName": "Smith",
    },
  },
  "html": "
<dl>
  <dt>First name</dt>
  <dd>Jordan</dd>
  <dt>Last name</dt>
  <dd>Smith</dd>
</dl>
",
}
`;

exports[`with setImmediate polyfill: after update firstName 1`] = `
Object {
  "Options": Object {
    "ID": "component",
    "user": Object {
      "firstName": "Jordan",
      "lastName": "Smith",
    },
  },
  "html": "
<dl>
  <dt>First name</dt>
  <dd>John</dd>
  <dt>Last name</dt>
  <dd>Smith</dd>
</dl>
",
}
`;

exports[`with setImmediate polyfill: after update lastName & wait updating 1`] = `
Object {
  "Options": Object {
    "ID": "component",
    "user": Object {
      "firstName": "Jordan",
      "lastName": "Shah",
    },
  },
  "html": "
<dl>
  <dt>First name</dt>
  <dd>Jordan</dd>
  <dt>Last name</dt>
  <dd>Smith</dd>
</dl>
",
}
`;

exports[`with setImmediate polyfill: after update lastName 1`] = `
Object {
  "Options": Object {
    "ID": "component",
    "user": Object {
      "firstName": "Jordan",
      "lastName": "Shah",
    },
  },
  "html": "
<dl>
  <dt>First name</dt>
  <dd>Jordan</dd>
  <dt>Last name</dt>
  <dd>Smith</dd>
</dl>
",
}
`;

exports[`with setImmediate polyfill: initial 1`] = `
Object {
  "Options": Object {
    "ID": "component",
    "user": Object {
      "firstName": "John",
      "lastName": "Smith",
    },
  },
  "html": "
<dl>
  <dt>First name</dt>
  <dd>John</dd>
  <dt>Last name</dt>
  <dd>Smith</dd>
</dl>
",
}
`;

exports[`with setImmediate polyfill: lastName don't changed 1`] = `
Object {
  "Options": Object {
    "ID": "component",
    "user": Object {
      "firstName": "Jordan",
      "lastName": "Shah",
    },
  },
  "html": "
<dl>
  <dt>First name</dt>
  <dd>Jordan</dd>
  <dt>Last name</dt>
  <dd>Shah</dd>
</dl>
",
}
`;
